tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature

```diff
@@ -0,0 +1,64 @@
+@api
+Feature: assign role
+  As an admin, I want to assign roles to users.
+  I cannot change my own role.
+  Users without an admin role cannot get the list of roles, assignments list and assign roles to users
+
+  Scenario Outline: only admin user can see all existing roles
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And the administrator has given "Alice" the role "<userRole>" using the settings api
+    When user "Alice" tries to get all existing roles
+    Then the HTTP status code should be "<statusCode>"
+    Examples:
+      | userRole    | statusCode |
+      | Admin       | 201        |
+      | Space Admin | 401        |
+      | User        | 401        |
+
+
+  Scenario Outline: only admin user can see assignments list
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And the administrator has given "Alice" the role "<userRole>" using the settings api
+    When user "Alice" tries to get asigments list
+    Then the HTTP status code should be "<statusCode>"
+    Examples:
+      | userRole    | statusCode |
+      | Admin       | 201        |
+      | Space Admin | 401        |
+      | User        | 401        |
+
+
+  Scenario Outline: a user cannot change own role
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And the administrator has given "Alice" the role "<userRole>" using the settings api
+    When user "Alice" changes his own role to "<desiredRole>"
+    Then the HTTP status code should be "400"
+    Examples:
+      | userRole    | desiredRole |
+      | Admin       | User        |
+      | Admin       | Space Admin |
+      | Space Admin | Admin       |
+      | Space Admin | Space Admin |
+      | User        | Admin       |
+      | User        | Space Admin |
+    And user "Alice" should have the role "<userRole>"
+
+
+  Scenario Outline: only admin user can change the role for another user
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+    And the administrator has given "Alice" the role "<userRole>" using the settings api
+    When user "Alice" changes the role "<desiredRole>" for user "Brian"
+    Then the HTTP status code should be "<statusCode>"
+    Examples:
+      | userRole    | desiredRole | statusCode | expectedRole |
+      | Admin       | User        | 201        | User         |
+      | Admin       | Space Admin | 201        | Space Admin  |
+      | Admin       | Admin       | 201        | Admin        |
+      | Space Admin | Admin       | 400        | User         |
+      | Space Admin | Space Admin | 400        | User         |
+      | User        | Admin       | 400        | User         |
+      | User        | Space Admin | 400        | User         |
+    And user "Brian" should have the role "<expectedRole>"
```

Comment:
This seems a bit odd. We usually put examples table at the scenario end. Never thought this would work :smile:

Reply:
Ahh, I didn't think about it. I created first table and after that I created `user should have the role` method